lib/ui/splash/login_screen.dart

```diff
@@ -0,0 +1,228 @@
+    curve: Curves.linear,
+  )..addStatusListener((status) {
+      if (status == AnimationStatus.completed) {
+        final shouldAnimateLogoPosition =
+            ref.read(_shouldAnimateLogoPositionProvider.notifier);
+        Future.delayed(_logoDuration, () {
+          shouldAnimateLogoPosition.state = !shouldAnimateLogoPosition.state;
+        });
+      }
+    });
+
+  late final AnimationController _loginFormRevealAnimationController =
+      AnimationController(
+    duration: _loginFormRevealDuration,
+    vsync: this,
+  );
+  late final Animation<double> _loginFormRevealAnimation = CurvedAnimation(
+    parent: _loginFormRevealAnimationController,
+    curve: Curves.easeIn,
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return _buildLoginScreen();
+  }
+
+  Widget _buildLoginScreen() {
+    return Scaffold(
+      resizeToAvoidBottomInset: false,
+      body: Stack(children: [
+        Consumer(builder: (_, widgetRef, __) {
+          final shouldAnimateBlur =
+              widgetRef.watch(_shouldAnimateLogoPositionProvider);
+          return DimmedImageBackground(
+            image: Assets.images.bgSplash.image().image,
+            shouldBlur: true,
+            shouldAnimate: shouldAnimateBlur,
+            animationDuration: _loginFormRevealDuration,
+          );
+        }),
+        Consumer(
+          builder: (_, widgetRef, __) {
+            final shouldAnimateLogoPosition =
+                widgetRef.watch(_shouldAnimateLogoPositionProvider);
+            return AnimatedPositioned(
+              duration: _loginFormRevealDuration,
+              curve: Curves.easeIn,
+              bottom: 0.0,
+              right: 0.0,
+              left: 0.0,
+              top: shouldAnimateLogoPosition ? -500 : 0.0,
+              child: Center(
+                child: AnimatedScale(
+                  duration: _loginFormRevealDuration,
+                  scale: shouldAnimateLogoPosition ? 0.83 : 1,
+                  curve: Curves.easeIn,
+                  child: FadeTransition(
+                    opacity: _logoRevealAnimation,
+                    child: SvgPicture.asset('assets/images/logo_white.svg'),
+                  ),
+                ),
+              ),
+            );
+          },
+        ),
+        Consumer(builder: (_, widgetRef, __) {
+          final shouldRevealLoginForm =
+              widgetRef.watch(_shouldAnimateLogoPositionProvider);
+          if (shouldRevealLoginForm) {
+            _loginFormRevealAnimationController.forward();
+          }
+          return Padding(
+            padding: const EdgeInsets.all(24),
+            child: FadeTransition(
+              opacity: _loginFormRevealAnimation,
+              child: const LoginForm(),
+            ),
+          );
+        })
+      ]),
+    );
+  }
+
+  @override
+  void dispose() {
+    _logoRevealAnimationController.dispose();
+    super.dispose();
+  }
+}
+
+class LoginForm extends ConsumerStatefulWidget {
```

Comment:
What do you think about moving this widget to a separate file? 🙏 
For example: `lib/ui/login/widgets/login_form.dart`

Reply:
Updated in https://github.com/nimblehq/ic-flutter-avishek/pull/33/commits/013d5f49facd12f481667505c58cc29314cd7e3d